src/main/java/com/google/api/codegen/transformer/ApiCallableTransformer.java

```diff
@@ -0,0 +1,160 @@
+      pagedApiCallableBuilder.responseTypeName(pagedResponseTypeName);
+      pagedApiCallableBuilder.name(context.getNamer().getPagedCallableName(method));
+      pagedApiCallableBuilder.settingsFunctionName(
+          context.getNamer().getSettingsFunctionName(method));
+
+      apiCallables.add(pagedApiCallableBuilder.build());
+    }
+
+    return apiCallables;
+  }
+
+  private List<ApiCallSettingsView> generateApiCallableSettings(MethodTransformerContext context) {
+    SurfaceNamer namer = context.getNamer();
+    ModelTypeTable typeTable = context.getTypeTable();
+    Method method = context.getMethod();
+    MethodConfig methodConfig = context.getMethodConfig();
+
+    ApiCallSettingsView.Builder settings = ApiCallSettingsView.newBuilder();
+
+    settings.methodName(namer.getApiMethodName(method));
+    settings.requestTypeName(typeTable.getAndSaveNicknameFor(method.getInputType()));
+    settings.responseTypeName(typeTable.getAndSaveNicknameFor(method.getOutputType()));
+    settings.grpcTypeName(
+        typeTable.getAndSaveNicknameFor(namer.getGrpcContainerTypeName(context.getInterface())));
+    settings.grpcMethodConstant(namer.getGrpcMethodConstant(method));
+    settings.retryCodesName(methodConfig.getRetryCodesConfigName());
+    settings.retryParamsName(methodConfig.getRetrySettingsConfigName());
+
+    String notImplementedPrefix = "ApiCallableTransformer.generateApiCallableSettings - ";
+    settings.resourceTypeName(
+        namer.getNotImplementedString(notImplementedPrefix + "resourceTypeName"));
+    settings.pageStreamingDescriptorName(
+        namer.getNotImplementedString(notImplementedPrefix + "pageStreamingDescriptorName"));
+    settings.bundlingDescriptorName(
+        namer.getNotImplementedString(notImplementedPrefix + "bundlingDescriptorName"));
+
+    if (methodConfig.isPageStreaming()) {
+      namer.addPageStreamingCallSettingsImports(typeTable);
+      settings.type(ApiCallableType.PagedApiCallable);
+      TypeRef resourceType = methodConfig.getPageStreaming().getResourcesField().getType();
+      settings.resourceTypeName(typeTable.getAndSaveNicknameForElementType(resourceType));
+      settings.pageStreamingDescriptorName(namer.getPageStreamingDescriptorConstName(method));
+    } else {
+      if (methodConfig.isBundling()) {
+        namer.addBundlingCallSettingsImports(typeTable);
+        settings.type(ApiCallableType.BundlingApiCallable);
+        settings.bundlingDescriptorName(namer.getBundlingDescriptorConstName(method));
+        settings.bundlingConfig(generateBundlingConfig(context));
+      } else {
+        settings.type(ApiCallableType.SimpleApiCallable);
+      }
+    }
+
+    settings.memberName(namer.getSettingsMemberName(method));
+    settings.settingsGetFunction(namer.getSettingsFunctionName(method));
+
+    return Arrays.asList(settings.build());
+  }
+
+  private BundlingConfigView generateBundlingConfig(MethodTransformerContext context) {
```

Comment:
It seems like this will be useful outside of this class in other languages that support bundling. I don't know where it could be placed at the moment that would make more sense though, maybe ApiMethodTransformer?